__WTF/Scripts/Weapons/Health.cs

```diff
@@ -174,9 +178,10 @@ public virtual void Damage(HealthEventArgs e)
 			{
 				Killed?.Invoke(this, e);
 				
-				if (m_destroyOnKilled)
+				if (m_destroyOnKilled && !m_dying)
 				{
-					GameObjectUtil.Destroy(gameObject);
+					GameObjectUtil.Destroy(gameObject, m_destroyDelay = 2f);
```

Comment:
No need for the default value of 2 here. It will always use the component variable

Reply:
done 👍